org.planqk.nisq.analyzer.core/src/main/java/org/planqk/nisq/analyzer/core/knowledge/prolog/PrologQueryEngine.java

```diff
@@ -205,7 +206,16 @@ private String assembleQueryForRule(String rule, Map<String, String> params, boo
             }
 
             // FIXME: avoid replacing parts of another variable where the name contains the searched variable, e.g., search for variable 'A' and replace part of variable 'AB' by accident
-            parameterPart = parameterPart.replaceFirst(variable, params.get(variable));
+
+            // Prepare the parameter for Prolog execution
+            String param = params.get(variable).trim();
+            if (!param.matches("-?\\s*[0-9]+\\.?[0-9]*"))
```

Comment:
Does not work if input is a binary **string**, e.g. 0110. NISQ Analyzer interprets it as an integer

Reply:
should be deleted!